src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateToRemoteDependencies.cs

```diff
@@ -25,17 +23,46 @@ public class UpdateToRemoteDependencies : BaseDependenciesTask
         /// </summary>
         public string GitHubAuthToken { get; set; }
         public string GitHubUser { get; set; }
-        public string GitHubEmail { get; set; }
+
+        [Output]
+        public bool MadeChanges { get; set; }
+
+        [Output]
+        public string SuggestedCommitMessage { get; set; }
 
         protected override void TraceListenedExecute()
         {
-            using (GitHubClient client = CreateClient(allowAnonymous: true))
-            {
-                DependencyUpdateResults updateResults = UpdateToRemote(client);
+            GitHubAuth auth = null;
 
+            if (string.IsNullOrEmpty(GitHubAuthToken))
+            {
                 Log.LogMessage(
                     MessageImportance.Low,
-                    $"Suggested commit message: '{updateResults.GetSuggestedCommitMessage()}'");
+                    $"No value for '{nameof(GitHubAuthToken)}'. " +
+                    "Accessing GitHub API anonymously.");
+            }
+            else
+            {
+                auth = new GitHubAuth(GitHubAuthToken, GitHubUser, null);
+            }
+
+            using (var client = new GitHubClient(auth))
```

Comment:
Should this use "CreateClient" below?

Reply:
Good catch, `CreateClient` should actually be removed. I didn't see enough overlap between `UpdateToRemoteDependencies` and `CreatePullRequest` to justify keeping it--the split simplified it quite a bit.